Messages/IDelete.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Messages
+{
+     interface IDelete
```

Comment:
In your example both classes  inherited Message use interface IDelete. So class Message should be inherit this interface)